File: c++/leetcode/editor/cn/P34_FindFirstAndLastPositionOfElementInSortedArray.cpp
#include <bits/stdc++.h>

using namespace std;

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
public:
    vector<int> searchRange(vector<int> &nums, int target) {
        if (nums.empty()) {
            return {-1, -1};
        }
        if (nums[0] > target || *(nums.end() - 1) < target) {
            return {-1, -1};
        }
        int left = 0;
        int right = nums.size();
        int l = -2;
        int r = -2;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] <= target) {

            }
        }
        left = 0;
        right = nums.size();
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] <= target) {
                left = mid + 1;
                r = left;
            } else {
                right = mid;
            }
        }
        if (r - l >= 1) {
            return {l, r - 1};
        }
        return {-1, -1};
    }
};
//leetcode submit region end(Prohibit modification and deletion)


int main() {
    Solution solution;
    vector v{5, 7, 7, 8, 8, 10};
//    vector v{1};
    solution.searchRange(v, 8);
}

File: python/leetcode/editor/cn/P34_FindFirstAndLastPositionOfElementInSortedArray.py
# leetcode submit region begin(Prohibit modification and deletion)
from typing import List


class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        if len(nums) == 0:
            return [-1, -1]
        left = 0
        right = len(nums) - 1
        res = []
        while left <= right:
            mid = left + (right - left) // 2
            if nums[mid] < target:
                left = mid + 1
            else:
                right = mid - 1
        res.append(left)
        left = res[0]
        if left == len(nums) or nums[left] != target:
            return [-1, -1]
        right = len(nums) - 1
        while left <= right:
            mid = left + (right - left) // 2
            if nums[mid] <= target:
                left = mid + 1
            else:
                right = mid - 1
        res.append(left - 1)

        return res


# leetcode submit region end(Prohibit modification and deletion)
s = Solution
# print(s.searchRange(s, [5, 7, 7, 8, 8, 10], 8))
print(s.searchRange(s, [5, 7, 7, 8, 8, 10], 6))
# print(s.searchRange(s, [], 6))
# print(s.searchRange(s, [2, 2], 3))
